chimney/src/main/scala/io/scalaland/chimney/ErrorPathNode.scala

```diff
@@ -0,0 +1,33 @@
+package io.scalaland.chimney
+
+sealed trait ErrorPathNode {
+  def show: String
+
+  def separator: String
+}
+
+object ErrorPathNode {
+  case class Accessor(name: String) extends ErrorPathNode {
+    def show: String = name
+
+    val separator: String = "."
+  }
+
+  case class Index(value: Int) extends ErrorPathNode {
+    def show: String = s"[$value]"
+
+    val separator: String = ""
+  }
+
+  case class MapKey[K](key: K) extends ErrorPathNode {
```

Comment:
Does `K` carry any reasonable type-level information when being element of `List[ErrorPathNode]`?
It seems it will be LUB-bed to AnyRef anyway, so I guess `key: AnyRef` may be sufficient here.

Reply:
Fixed